src/server/services/procedures/new-york-public-library/new-york-public-library.js

```diff
@@ -40,6 +40,16 @@ NYPL.search = async function(term, perPage = 50, page = 1) {
     return ret;
 };
 
+NYPL._searchForKey = function(json, key) {
+    if (json instanceof Array) {
```

Comment:
I believe this entire function could be simplified to 
```javascript
        for (const item of listify(json)) {
            if (key in item) return item[key];
        }
        return undefined;
```

Reply:
Honestly, there is still a bit of code duplication that we could take care of by making this function a little more abstract. I would probably make two functions: `getOriginField(originInfo, fieldName)` and `getFirstValue(dicts, key)`.

`getOriginField` could use `getFirstValue(listify(originInfo), fieldName)` then handle the duplicate logic which checks for `$` in the resultant object. There would still need to be some special logic for handling the location but this would at least condense the other two.

Reply:
Yeah that first comment checks out - didn't notice that. As for the code dupe part, I've added a new commit to address it, but it does so in a different way. It does combine those two areas into same logic and one line each.